### PR TITLE
Add Algolia Search Feature

### LINKED ISSUE #26: Search Feature
Right now, the search bar is there for styling but we should add the logic to search posts through [react-instantsearch](https://github.com/algolia/react-instantsearch/)

### PR DESCRIPTION
**Pull request by Kn99HN**

### CODE REVIEW

File: src/components/Header.js
Comment (esilverm) [inline src/components/Header.js]: Why are we removing this? There really isn't a reason to go back to the landing page

**Comment by Kn99HN** [comment]
Hmmmmm, Not sure why netlify is not building properly. Any idea @esilverm? People seems to run into similar problem with the algolia plugin so I have looked into that but it still doesn't build.

File: src/components/SearchBar/StyledSearchResult.js
Comment (esilverm) [inline src/components/SearchBar/StyledSearchResult.js]: We should be using the utils theme.js file to control this. Setting it up any other way will cause flashing problems. The css variables are loaded when the page is so there should be no problem here

File: src/components/SearchBar/index.js
Comment (esilverm) [inline src/components/SearchBar/index.js]: Mentioned this in the other file, but no reason this should be used outside of controlling the state of the theme

File: src/components/SearchBar/useClickOutside.js
Comment (esilverm) [inline src/components/SearchBar/useClickOutside.js]: This file should probably be within the utils folder in case we want to use it for other components

**Review by esilverm — COMMENTED** [review]
A few changes needed before i give the approval stamp. Also we gotta make site layout responsive sooner than later. Who wants to do it

**Comment by esilverm** [comment]
> Hmmmmm, Not sure why netlify is not building properly. Any idea @esilverm? People seems to run into similar problem with the [algolia plugin](https://github.com/algolia/gatsby-plugin-algolia/issues/44) so I have looked into that but it still doesn't build.

Looks like it has to do with your dot env file. Typically when im working with them I like to create examples named `.env-example`

with the variables empty so people can just plug it in and rename the file or whatever. How are we handling env variables on netlify?

**Comment by esilverm** [comment]
Also Happy April Fools Day 😂 😂 

Screen Shot 2021-04-01 at 12 25 59 PM

**Comment by Kn99HN** [comment]
> Looks like it has to do with your dot env file. Typically when im working with them I like to create examples named `.env-example`

I'm setting it in `Environment Variables` under `Build & Deploy`. It's supposed to work based on the [blog](https://www.gatsbyjs.com/docs/adding-search-with-algolia/#deploying-to-netlify) I believe...

File: README.md
Comment (esilverm) [inline README.md]:
```suggestion
ALGOLIA_ADMIN_KEY=xxxxx
```

**Comment by esilverm** [comment]
> > Looks like it has to do with your dot env file. Typically when im working with them I like to create examples named `.env-example`
> 
> I'm setting it in `Environment Variables` under `Build & Deploy`. It's supposed to work based on the [blog](https://www.gatsbyjs.com/docs/adding-search-with-algolia/#deploying-to-netlify) I believe...

Build works just had to clear cache so env variables get loaded

**Comment by esilverm** [comment]
While you're at it can you actually go ahead and update gatsby, just use the `yarn upgrade-interactive` command and upgrade all

**Comment by esilverm** [comment]
Can you try making that change and pushing again, I removed the plugin that may be causing this issue.

**Comment by Kn99HN** [comment]
> While you're at it can you actually go ahead and update gatsby, just use the `yarn upgrade-interactive` command and upgrade all

I thought we are using npm?

**Comment by esilverm** [comment]
> > While you're at it can you actually go ahead and update gatsby, just use the `yarn upgrade-interactive` command and upgrade all
> 
> I thought we are using npm?

Are you not using `yarn` to manage your dependencies for this project??? How did you manage to change the yarn.lock file otherwise???

**Comment by Kn99HN** [comment]
> > > While you're at it can you actually go ahead and update gatsby, just use the `yarn upgrade-interactive` command and upgrade all
> > 
> > 
> > I thought we are using npm?
> 
> Are you not using `yarn` to manage your dependencies for this project??? How did you manage to change the yarn.lock file otherwise???

Lolllllllll 

~Anw, just gatsby right, and no other dependencies or do you want to upgrade everything including plugins?~ Nvm didn't read your message carefully. I'll upgrade them and push

**Review by esilverm — APPROVED** [review]

**Comment by esilverm** [comment]
We should fix styling but that can come later. This should be fine to merge imo

**Comment by Kn99HN** [comment]
> We should fix styling but that can come later. This should be fine to merge imo

Yeah I was about to say that. Styling for the searchResultBox is not great atm but the logic works for now.

**Pull request merged**
